src/dwg.c

```diff
@@ -3192,3 +3205,230 @@ dwg_log_dataflags (const int _loglevel, const int maxlevel,
         HANDLER (OUTPUT, "  vert_alignment (128)\n");
     }
 }
+
+static int
+Dwg_Object_Ref_compare (const Dwg_Object_Ref *key, const Dwg_Object_Ref **ppR)
+{
+  int retVal = (int)key->handleref.code - (int)(*ppR)->handleref.code;
+  if (0 != retVal)
+    {
+      return retVal;
+    }
+  return (long)key->absolute_ref - (long)(*ppR)->absolute_ref;
+}
+
+// from M$ bsearch.c
+/***
+ *void *bsearch_ex() - do a binary search on an array
+ *
+ *Purpose:
+ *   Does a binary search of a sorted array for a key.
+ *
+ *Entry:
+ *   const void *key    - key to search for
+ *   const void *base   - base of sorted array to search
+ *   size_t num         - number of elements in array
+ *   size_t width       - number of bytes per element
+ *   int (*compare)(const void* item1,const void* item2) - pointer to function
+ *that compares two array elements, returning neg when #1 < #2, pos when #1 >
+ *#2, and 0 when they are equal. Function is passed pointers to two array
+ *elements. const void** ppBefore - pointer to key's insert pos in array,if key
+ *not exist if *ppBefore = NULL,key should append at tail if ppBefore = NULL,
+ *caller don't care about insert pos.
+ *
+ *Exit:
+ *   if key is found:
+ *           returns pointer to occurrence of key in array
+ *           *ppBefore = NULL
+ *   if key is not found:
+ *           returns NULL
+ *           *ppBefore pointer to key's insert pos in array
+ *
+ *Exceptions:
+ *   Input parameters are validated. Refer to the validation section of the
+ *function.
+ *
+ *******************************************************************************/
+static void *
+bsearch_ex (const void *key, const void *base, size_t num, size_t width,
```

Comment:
bsearch is in C99 already. Is this needed for VS2019?

Reply:
I need bsearch_ex to return the insertion point when the key is not found.

Reply:
Oh my, well then